index.html

```diff
@@ -465,6 +472,28 @@
 						$('#root_firmware_size').css('borderColor', '');
 					}
 				}
+
+				// if information is unavailble this prevents issues with jquery
+				if (data.hasOwnProperty("firmware"))
+				{
+					// remove old firmware button
+					$('#button_firmware_githubUrl').remove();
+					// reshow firmware field incase we don't replace it
```

Comment:
Wouldn't it just be simpler to leave the value blank if nothing there (as it currently does) and then always turn this field into a button/link? If empty there is nothing to click, otherwise whatever is populated is clickable.

I'm just worried about adding too much technical debt with these fancy HTML overrides etc.

Reply:
i didn't like the idea of leaving it blank as someone might think something is wrong ; 
thus it instead will show not available.
the way it is set up however the button element will only ever showup / be clickable when the URL is actually there.
the line 480 is the same type of line used in the icon code which prevents old injections from duplicating.